Unable to start application context from Android due to Regex

## Steps to reproduce: 

- Create a new Android application. 
- Set min SDK to 24. 
- Add several `packagingOptions` exclusions.
- Add Micronaut dependencies: 

_app/build.gradle_

```gradle
...
android {
    compileSdkVersion 29

    defaultConfig {
        applicationId "example.micronaut"
        minSdkVersion 24
        ...
    }
    ...
   packagingOptions {
        exclude 'META-INF/config-properties.adoc'
        exclude 'META-INF/spring-configuration-metadata.json'
    }
}
...
dependencies {
    annotationProcessor "io.micronaut:micronaut-inject-java:2.5.4"
    compileOnly "io.micronaut:micronaut-inject-java:2.5.4"
    implementation "io.micronaut:micronaut-runtime:2.5.4"
...
..
```
Add an [Android Application](https://developer.android.com/reference/android/app/Application) class : 

```java
package example.micronaut;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import io.micronaut.context.ApplicationContext;
import io.micronaut.context.env.Environment;

public class BaseApplication extends Application {

    ApplicationContext ctx;


    public BaseApplication() {
        super();
    }

    @Override
    public void onCreate() {
        super.onCreate();

        ctx = ApplicationContext.builder(MainActivity.class, Environment.ANDROID).start();
        registerActivityLifecycleCallbacks(new ActivityLifecycleCallbacks() {
            @Override
            public void onActivityCreated(@NonNull Activity activity, @Nullable Bundle bundle) {
                ctx.inject(activity);
            }

            @Override
            public void onActivityStarted(@NonNull Activity activity) {

            }

            @Override
            public void onActivityResumed(@NonNull Activity activity) {

            }

            @Override
            public void onActivityPaused(@NonNull Activity activity) {

            }

            @Override
            public void onActivityStopped(@NonNull Activity activity) {

            }

            @Override
            public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle bundle) {

            }

            @Override
            public void onActivityDestroyed(@NonNull Activity activity) {

            }
        });
    }

    @Override
    public void onTerminate() {
        super.onTerminate();
        if(ctx != null && ctx.isRunning()) {
            ctx.stop();
        }
    }
}
```

Modify `src/main/AndroidManifest.xml` to use `BaseApplication`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="example.micronaut">

    <application
        android:name=".BaseApplication"
...
..
.
```

Run the application:

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: example.micronaut, PID: 8208
    java.lang.ExceptionInInitializerError
        at io.micronaut.context.DefaultApplicationContext.createEnvironment(DefaultApplicationContext.java:155)
        at io.micronaut.context.DefaultApplicationContext.<init>(DefaultApplicationContext.java:107)
        at io.micronaut.context.DefaultApplicationContextBuilder.newApplicationContext(DefaultApplicationContextBuilder.java:294)
        at io.micronaut.context.DefaultApplicationContextBuilder.build(DefaultApplicationContextBuilder.java:251)
        at io.micronaut.context.ApplicationContextBuilder.start(ApplicationContextBuilder.java:224)
        at example.micronaut.BaseApplication.onCreate(BaseApplication.java:25)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1119)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5740)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.util.regex.PatternSyntaxException: Syntax error in regexp pattern near index 80
    \$\{\s?random\.(\S+?)((\(-?\d+(\.\d+)?\))|(\[-?\d+(\.\d+)?,\s?-?\d+(\.\d+)?]))?}
                                                                                    ^
        at java.util.regex.Pattern.compileImpl(Native Method)
        at java.util.regex.Pattern.compile(Pattern.java:1340)
        at java.util.regex.Pattern.<init>(Pattern.java:1324)
        at java.util.regex.Pattern.compile(Pattern.java:946)
        at io.micronaut.context.env.PropertySourcePropertyResolver.<clinit>(PropertySourcePropertyResolver.java:60)
        	... 16 more
```

--- a/inject/src/main/java/io/micronaut/context/env/PropertySourcePropertyResolver.java
+++ b/inject/src/main/java/io/micronaut/context/env/PropertySourcePropertyResolver.java
@@ -57,7 +57,7 @@ public class PropertySourcePropertyResolver implements PropertyResolver {
     private static final String RANDOM_PREFIX = "\\s?random\\.(\\S+?)";
     private static final String RANDOM_UPPER_LIMIT = "(\\(-?\\d+(\\.\\d+)?\\))";
     private static final String RANDOM_RANGE = "(\\[-?\\d+(\\.\\d+)?,\\s?-?\\d+(\\.\\d+)?])";
-    private static final Pattern RANDOM_PATTERN = Pattern.compile("\\$\\{" + RANDOM_PREFIX + "(" + RANDOM_UPPER_LIMIT + "|" + RANDOM_RANGE + ")?}");
+    private static final Pattern RANDOM_PATTERN = Pattern.compile("\\$\\{" + RANDOM_PREFIX + "(" + RANDOM_UPPER_LIMIT + "|" + RANDOM_RANGE + ")?\\}");
     private static final char[] DOT_DASH = new char[] {'.', '-'};
     private static final Object NO_VALUE = new Object();
     private static final PropertyCatalog[] CONVENTIONS = {PropertyCatalog.GENERATED, PropertyCatalog.RAW};

